doc/md/fundamentals/2-types/3-functions.md

```diff
@@ -49,110 +46,123 @@ Motoko provides different types of functions based on where in the program they
-Since public functions must be `shared`, the `shared` keyword is optional and can be committed.  
+The public functions of an actor determine its external interface. All public functions in an actor must be shared and can be either `shared`, `shared query` or `shared composite query` functions. Private functions cannot be `shared`.
+Since an actor's public functions must be shared, the `shared` keyword is optional and can be omitted.  
 
-`shared` functions permanently update the state of an actor, while `query` and `composite query` functions are only executed for their result. While queries can temporarily alter the state of the actor, the changes are not permanent and never visible to other callers. It's as if each query operated on a copy of the actor that is discarded on return from the query. When called from a front-end, `query` functions have much lower latency than calling the equivalent `shared` method. This is because `shared` functions require the protocol to reach consensus on the state changes and result of a shared function, while `query` functions do not.  
+`shared` functions permanently update the state of an actor, while `query` and `composite` `query` functions are only executed for their result.
 
-Shared functions can be called from outside the canister (e.g., by users or other [canisters](https://internetcomputer.org/docs/building-apps/essentials/canisters)). These are processed [asynchronously](https://internetcomputer.org/docs/motoko/fundamentals/actors-async#async--await) because they interact with the network.
+Although queries can temporarily alter the state of an actor, these changes are not permanent and are never visible to other callers. It's as if each query operates on a copy of the actor, which is discarded when the query returns.
 
-```motoko no-repl  
-persistent actor Account {  
-  var balance = 0;  
-  
-  public shared func deposit(amount : Nat) : async Nat {  
-     balance += amount;  
-     balance  
-  }  
-}  
+When called from a front-end, `query` functions generally have much lower latency than equivalent shared functions. This is because shared functions require the protocol to reach consensus on the state changes and results, whereas query functions do not.
+
+```motoko no-repl
+persistent actor Account {
+  var balance = 0;
+
+  public shared func deposit(amount : Nat) : async Nat {
+    balance += amount;
+    balance
+  }
+}
 ```
 
 Omitting the shared keyword, we can also write:
 
-```motoko no-repl  
-persistent actor Account {  
-  var balance = 0;  
-  
-  public shared func deposit(amount : Nat) : async Nat {  
-     balance += amount;  
-     balance  
-  }  
-}  
+```motoko no-repl
+persistent actor Account {
+  var balance = 0;
+
+  public func deposit(amount : Nat) : async Nat {
+    balance += amount;
+    balance
+  }
+}
 ```
 
 The deposit function has type `: shared Nat -> async Nat`.
 
```

Comment:
It would be good to add an example:

````suggestion
Consider this code:

``` motoko include=Account
let b1 = await Account.deposit(50);
let b2 = await Acccount.deposit(50);
(b1,b2)
```
The first class to `Account.balance(50)` increments `balance` from `0` to `50`, returning `50`.
The second call increments the balance from `50` to `100`, returning `100`.

Because `Account.deposit` is asynchronous, its results are returned in futures of type `async Nat`. Calling `await` on a each future extracts the results of the calls when they become available (so `b1` is `50` and `b2` is `100`).
````